### ISSUE TITLE
Cannot reconstruct noise using ddim_inverse scheduler.

### ISSUE THREAD

**Issue by yanzengatsmu** [huggingface/diffusers#5703]
### Describe the bug

I'm currently trying to replicate the experiments in DDIM( 5.4 RECONSTRUCTION FROM LATENT SPACE), but the results are quite different from what's reported in the paper. I wanted to ask if you've also conducted experiments on inverting image back into noise, and what were the results?

### Reproduction
First generate a image using ddim and save the final image tensor in 'original.pt'.
Second reconsturct the noise 'reverse.pt' from 'original.pt'
Finally generate using 'reverse.pt'.
The code in pipeline_ddim.py will be modified during the process.

reconstruction part:
from diffusers import DDIMPipeline, DDIMInverseScheduler
model_id = "google/ddpm-cifar10-32"
ddim = DDIMPipeline.from_pretrained(model_id).to('cuda')   
ddim.scheduler = DDIMInverseScheduler.from_config(ddim.scheduler.config)
img = ddim(num_inference_steps=500).images[0]  

### System Info

- `diffusers` version: 0.21.2
- Platform: Linux-5.4.0-150-generic-x86_64-with-glibc2.27
- Python version: 3.11.5
- PyTorch version (GPU?): 2.0.1 (True)
- Huggingface_hub version: 0.17.3
- Transformers version: 4.33.2
- Accelerate version: 0.23.0
- xFormers version: not installed
- Using GPU in script?: no
- Using distributed or parallel set-up in script?: no

**Comment by patrickvonplaten** [comment]
I don't understand this code snippet:

```py
from diffusers import DDIMPipeline, DDIMInverseScheduler
model_id = "google/ddpm-cifar10-32"
ddim = DDIMPipeline.from_pretrained(model_id).to('cuda')
ddim.scheduler = DDIMInverseScheduler.from_config(ddim.scheduler.config)
img = ddim(num_inference_steps=500).images[0]
```

I think you have to create a DDIMPipeline that allows to take original.pt as an input

**Comment by yanzengatsmu** [comment]
Sorry for the confusion. Let me explain in detail. The experiment consist of three parts.

1. Generating Image: In this step, I generate an image using code below. Also I modified the DDIMPipeline to save the tensor in the last denoising step.
`from diffusers import DDIMPipeline
import torch
model_id = "google/ddpm-cifar10-32"
ddim = DDIMPipeline.from_pretrained(model_id).to('cuda')
img = ddim(num_inference_steps=500).images[0]
img.save("original.png")`
2. Reversing DDIM: In this step, I modified the DDIMPipeline again by replacing the randn_tensor function with torch.load which loads the tensor in first step. Also, the tensor in last denoising step was saved. But I am not sure if I used DDIMInverseScheduler correctly. The code is like below.
`from diffusers import DDIMPipeline, DDIMInverseScheduler
model_id = "google/ddpm-cifar10-32"
ddim = DDIMPipeline.from_pretrained(model_id).to('cuda')
ddim.scheduler = DDIMInverseScheduler.from_config(ddim.scheduler.config, num_train_timesteps=1000)
img = ddim(num_inference_steps=500).images[0]`
3. Generating Image Using the Reversed Tensor: Finally, I use the tensor in step two to generate an image again. 
`from diffusers import DDIMPipeline
import torch
model_id = "google/ddpm-celebahq-256"
ddim = DDIMPipeline.from_pretrained(model_id).to('cuda')
img = ddim(num_inference_steps=50).images[0]
img.save("reverse.png")`

The result shows that the reversed tensor cannot reconsturct the image generated in step one.

**Comment by yanzengatsmu** [comment]
Solved. The problem is caused by the wrong setting of parameters of scheduler.

**Comment by david20571015** [comment]
> Solved. The problem is caused by the wrong setting of parameters of scheduler.

Sorry to reopen this issue, @yanzengatsmu. I'm experiencing a reconstruction problem similar to yours. Could you please tell me how you solved it? Thanks!

**Comment by yanzengatsmu** [comment]
> > Solved. The problem is caused by the wrong setting of parameters of scheduler.
> 
> Hello, @yanzengatsmu. I'm experiencing a reconstruction problem similar to yours. Could you please tell me how you solved it? Thanks!

It has been a while, so I am sorry I don't remember all the detail. I think the parameter, "clip_sample", in scheduler plays a important role in this process. You can try to set it to false.

**Comment by david20571015** [comment]
> > > Solved. The problem is caused by the wrong setting of parameters of scheduler.
> > 
> > 
> > Hello, @yanzengatsmu. I'm experiencing a reconstruction problem similar to yours. Could you please tell me how you solved it? Thanks!
> 
> It has been a while, so I am sorry I don't remember all the detail. I think the parameter, "clip_sample", in scheduler config plays a important role in this process. You can try to set it to false.

Thank you! It works well.
